kernel/task.h

```diff
@@ -47,6 +47,16 @@ struct task {
     sigset_t_ saved_mask;
     bool has_saved_mask;
 
+    struct {
+        bool stopped;
+        bool reported_stop;
+        cond_t stop_cond;
+        lock_t stop_lock;
```

Comment:
I think this should be named just `lock`, so `ptrace.lock` protects all the things that can be modified through ptrace. Also add a comment saying what it locks